packages/runtime/src/components/page/HeadSnippet.tsx

```diff
@@ -0,0 +1,122 @@
+import { Children, createElement, useEffect, useRef } from 'react'
+import { Snippet } from '../hooks/usePageSnippets'
+import { useServerInsertedHTML } from 'next/navigation'
+import parse from 'html-react-parser'
+import Head from 'next/head'
+import { useIsPagesRouter } from '../../next/hooks/use-is-pages-router'
+
+type Props = {
+  snippet: Snippet
+}
+
+export function HeadSnippet({ snippet }: Props) {
+  const isPagesRouter = useIsPagesRouter()
+
+  useEffect(() => {
+    if (isPagesRouter) {
+      return () => {
+        const snippetElements = getSnippetElementsFromDOM(snippet)
+        if (snippetElements.length > 0) return
+        cleanUpSnippet(snippet)
```

Comment:
Problem: `next/head` will dedupe scripts rendered in the head. Even though the page mounts/unmounts on client-side navigations, the script will not actually execute twice. It just stays in the head. However, the page mounting/unmounting means that we perform the cleanup, but the snippet never reruns. Checking the presence of the snippet prevents this.